sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/LNAdapterTest.java

```diff
@@ -505,33 +504,81 @@ void updateDAI_should_not_update_DAI_Val_when_DTT_Fc_not_defined() {
                 .getDOIAdapterByName("Do")
                 .getStructuredDataAdapterByName("sdo1")
                 .getStructuredDataAdapterByName("d")
+                .getStructuredDataAdapterByName("antRef")
                 .getStructuredDataAdapterByName("bda1")
                 .getStructuredDataAdapterByName("bda2")
-                .getStructuredDataAdapterByName("bda3")
-                .getDataAdapterByName("antRef");
+                .getDataAdapterByName("bda3");
 
         assertThat(daiAdapter.getCurrentElem().getVal().get(0).getValue()).isEqualTo("Completed-diff");
 
         System.out.println(MarshallerWrapper.marshall(scd));
     }
 
     @Test
-    @Disabled(value = "Disable while bug #241 is not fixed")
-    /**
-     * @see <a href="https://github.com/com-pas/compas-sct/issues/241" target="_blank">Issue !241 (UpdateDAI Val does not produce subelements (SDI) as expected)</a>
-     */
-    void updateDAI_should_update_DAI_values_when_data_updatable() {
+    void updateDAI_should_update_DOI_SDI_DAI_that_exists_when_data_updatable() {
         // Given
         ResumedDataTemplate rDtt = new ResumedDataTemplate();
         SCL scd = SclTestMarshaller.getSCLFromFile("/ied-test-schema-conf/ied_unit_test.xml");
-        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        IEDAdapter iAdapter = assertDoesNotThrow(() -> sclRootAdapter.getIEDAdapterByName("IED_NAME"));
-        LDeviceAdapter lDeviceAdapter = assertDoesNotThrow(() -> iAdapter.findLDeviceAdapterByLdInst("LD_INS2").get());
-        LN0Adapter lnAdapter = (LN0Adapter) AbstractLNAdapter.builder()
-                .withLDeviceAdapter(lDeviceAdapter)
-                .withLnClass(TLLN0Enum.LLN_0.value())
-                .build();
+        AbstractLNAdapter<?> lnAdapter = findLn0(new SclRootAdapter(scd), "IED_NAME", "LD_INS1");
+
+        rDtt.setDoName(new DoTypeName("Do.sdo1.d"));
+        rDtt.setDaName(new DaTypeName("antRef.bda1.bda2.bda3"));
+        rDtt.setFc(TFCEnum.SE);
+        rDtt.setValImport(true);
+        assertThat(rDtt.isUpdatable()).isTrue();
+        rDtt.setDaiValues(List.of(newVal("newValue")));
+
+        // When
+        lnAdapter.updateDAI(rDtt);
+
+        // Then
+        System.out.println(MarshallerWrapper.marshall(scd));
+        SDIAdapter.DAIAdapter daiAdapter = lnAdapter
+                .getDOIAdapterByName("Do")
+                .getStructuredDataAdapterByName("sdo1")
+                .getStructuredDataAdapterByName("d")
+                .getStructuredDataAdapterByName("antRef")
+                .getStructuredDataAdapterByName("bda1")
+                .getStructuredDataAdapterByName("bda2")
+                .getDataAdapterByName("bda3");
```

Comment:
Only object of type `TDAI` can have value and `TSDI` can only have `TSDI` or `TDAI` .
The expected result should be: 
```java
SDIAdapter.DAIAdapter daiAdapter = lnAdapter
               .getDOIAdapterByName("Do") //name of TDOI
                .getStructuredDataAdapterByName("sdo1") //name of TSDI
                .getStructuredDataAdapterByName("d") //name of TSDI
                .getStructuredDataAdapterByName("bda1") //name of TSDI
                .getStructuredDataAdapterByName("bda2") //name of TSDI
                .getStructuredDataAdapterByName("bda3") //name of TSDI
                .getDataAdapterByName("antRef");//name of TDAI
assertThat(daiAdapter.getCurrentElem().getName()).isEqualTo("antRef");
assertThat(daiAdapter.getCurrentElem().getVal().get(0).getValue()).isEqualTo("newValue");
```
As mentioned in https://github.com/com-pas/compas-sct/issues/241
```xml
<DOI name="doName">
    <SDI name="sdoName1">
        <SDI name="sdoName2">
            <SDI name="bdaName1">
                <SDI name="bdaName2">
                    <SDI name="bdaName3">
                        <DAI name="daName" valImport="true">
                            <Val>newValue</Val>
                        </DAI>
                    </SDI>
                </SDI>
            </SDI>
        </SDI>
</DOI>
```